dm/worker/config.go

```diff
@@ -256,8 +260,50 @@ func (c *Config) configFromFile(path string) error {
 	return c.verify()
 }
 
+func (c *Config) adjust() error {
+	c.From.Adjust()
+	c.Checker.adjust()
+
+	fromDB, err := c.createBaseDB()
```

Comment:
can we create DB only needed (no `server-id`/ `flavor` set)?
but that may make the code ugly?